pkg/asset/installconfig/basedomain.go

```diff
@@ -21,6 +22,7 @@ import (
 
 type baseDomain struct {
 	BaseDomain string
+	Publish    types.PublishingStrategy
```

Comment:
This could be ok. 
This creates an informal dependency for the Publishing Strategy on BaseDomain. Typically such dependencies would be formalized in the asset graph. 

PublishingStrategy is a field in the Install Config, which depends on the BaseDomain. So the dependency chain seems theoretically fine.